notebooks/curation_api/python_raw/create_dataset_from_local_file.ipynb

```diff
@@ -0,0 +1,302 @@
+   "id": "f73c969a",
+   "metadata": {},
+   "source": [
+    "<font color='#bc00b0'>(Required) Provide the path to your api key file</font>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "07a2424b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "api_key_file_path = \"path/to/api-key.txt\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3e84d52e",
+   "metadata": {},
+   "source": [
+    "<font color='#bc00b0'>(Required) Provide the absolute path to the h5ad datafile to upload</font>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ce19b8b0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "filename = \"/absolute/path/to-datafile.h5ad\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "104aea94",
+   "metadata": {},
+   "source": [
+    "<font color='#bc00b0'>(Required) Enter your chosen `identifier` (see 'identifier' behavior rules in heading above) which will serve as a unique identifier _within this Collection_ for the resultant Dataset. **Must possess the '.h5ad' suffix**.</font>\n",
+    "    \n",
+    "When using curator tags, we recommmend using a tagging scheme that 1) makes sense to you, and 2) will help organize and facilitate your \n",
+    "automation of future uploads for adding new Datasets and replacing existing Datasets. Remember that curator tags can be used as the identifier when _adding or replacing_ Datasets, whereas Dataset id's (uuid's) can only be used as the identifier when _replacing_ Datasets."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a4e9b33a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "identifier = \"arbitrary/tag/chosen-by-you\"  # Or use the dataset id \"01234567-89ab-cdef-0123-456789abcdef\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "463870f1",
+   "metadata": {},
+   "source": [
+    "<font color='#bc00b0'>(Required) Enter the id of the Collection to which you wish to add this datafile as a Dataset</font>\n",
+    "\n",
+    "_The Collection id can be found by looking at the url path in the address bar \n",
+    "when viewing your Collection in the UI of the Data Portal website:_ `collections/{collection_id}`_. You can only add/replace Datasets in private Collections or private revisions of published Collections. In order to edit a published Collection, you must first create a revision of that Collection._"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "86483731",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "collection_id = \"01234567-89ab-cdef-0123-456789abcdef\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c1cf4dae",
+   "metadata": {},
+   "source": [
+    "### Import dependencies"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "80427063",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import boto3\n",
```

Comment:
why is the dependecy import so low in the script. Move it closer to the top where it is most common

Reply:
I did that because I view the dependency import as not being in the category of "instructional field fill-in", which is what I wanted to have up-front, after the initial headings. 

Everything the curator really needs to pay attention to should come first. Then the dependencies can be imported.

I agree that this breaks typical code file conventions but this is also notebook code, not for a production service. Different goal

Reply:
If anyone has ever written in python they know imports come as early as possible. If they are looking at these notebooks, then they have written some python. I don't your argument as a compelling enough reason to break convention. Knowing you don't have the libraries required to run the script is helpful to know at the beginning No one will be confused by having import at the top. You can even put the imports right after the initial docstring.

Reply:
Ok I'll make the reorder